### DISCUSSION TITLE
Multiple root elements detected. This is not supported

### DISCUSSION THREAD

**Discussion by Hesammousavi** [livewire/livewire#3814] [7. Bugs (archived)] [answered]
hi everybody
```
livewire.js?id=21fa1dd78491a49255cd:12 Livewire: Multiple root elements detected. This is not supported. See docs for more information https://laravel-livewire.com/docs/2.x/rendering-components#returning-blade header
```
i dont have this warning error in my localhost
but when i push, i got these errors and i cant figure out why ! cuz everything is same between local and host !
look at console log "https://roocket.ir"

**Comment by Hesammousavi** [comment]
@joshhanley

**Comment by joshhanley** [comment]
@Hesammousavi I had a quick look at your site, and it seems the html comment that Livewire uses to detect the end of a component is missing.

Are you running one or multiple servers? If so, it seems like something is serving the latest javascript assets, but maybe not the latest php code if the comments aren't being injected properly? Also can you confirm what patch version of Livewire you are using?

If you open your site locally, and inspect the code, you should see next to each Livewire component a `html` comment like `<!-- Livewire Component wire-end:<component-id> -->`. Can you see them locally?

**Comment by Hesammousavi** [comment]
@joshhanley thanks for your response
yeah i can see  in locally but it's not in server
my site runing on laravel forge
php 8.0
livewire 2.6.5

**Comment by Hesammousavi** [comment]
it's so wired 
i put this command into composer.json
" @php artisan vendor:publish --force --tag=livewire:assets --ansi" and it's must be update anytime i run `composer update`!
why can i see  in my local but not in server :/ wtf

**Reply by joshhanley** [reply]
@Hesammousavi hmm yeah that is really strange! I wonder why the server wouldn't be picking it up as it's actually php code that implements the comment insertion (so it should show up on render even if the assets are out of date).

Maybe try doing a `composer dump-autoload` and `php artisan optimize:clear` on your server and see if that helps.

If not, I would try either using composer to remove Livewire on your server and reinstall it again, or you could remove the vendor directory completely and doing a fresh `composer install`.

**Reply by Hesammousavi** [reply]
i tried every way you can imagine 😄 
i removed venodor and reinstall it with `composer install` totally fresh. it is really strange to me too 😄

**Reply by joshhanley** [reply]
@Hesammousavi oh man, I wonder what is going on! Seems so bizarre it wouldn't be running the code properly (especially if it works locally!)

**Reply by jeffchown** [reply]
Did you ever resolve this issue?  I'm now having the same problem with one of my apps.  

All is fine on localhost, but on both my dev and production servers, I'm receiving two `Livewire: Multiple root elements detected` errors - generated by my Livewire Navbar component and its nested Livewire UserIndicator component.

All of the `wire-end` comments are missing on the dev and production servers.

Localhost (valet) and both servers (created/configured by Forge) are running Livewire 2.7.2 and are running the exact same app codebase.  I have not published livewire assets locally and I have `php artisan view:clear`ed on all servers - no change.

Any help is/ideas are much appreciated.

**Reply by joshhanley** [reply]
@jeffchown have a look at this discussion #4015 as to why it might be throwing the warnings in production (and not locally). I'd be interested to know if your issue is the same.

They are just console warnings though, so it shouldn't impact the functioning of the app, as such can be ignored until it is fixed. There is also an open PR for a potential fix #4153.

Hope this helps!

**Reply by jeffchown** [reply]
Thanks, @joshhanley - I do use Cloudflare for most/all of my client domains/sites.

I tried changing the auto-minify settings (per screenshot) and, lo and behold, my production and dev servers now serve HTML that contains the `wire-end` comments!  

![image](https://user-images.githubusercontent.com/42819001/143334155-ce2f53d4-ca19-485f-b5eb-3714bb37924a.png)

Thanks to @hein-p (from thread https://github.com/livewire/livewire/discussions/4015), and to @joshhanley for bringing it to my attention!

**Reply by joshhanley** [reply]
@jeffchown no worries! Glad you got it sorted 🙂

**Comment by Hesammousavi** [comment, marked as answer]
https://github.com/livewire/livewire/discussions/3814#discussioncomment-1697356
your answer totaly correct thank you man 

@jeffchown
